catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java

```diff
@@ -140,41 +145,138 @@
+                    multithreaded,
+                    0L,
+                    TimeUnit.MILLISECONDS,
+                    blockingQueue,
+                    rejectedExecutionHandler);
+            console.printf("Running a maximum of %d threads during replication.%n", multithreaded);
+
+            printProgressAndFlush(start, totalWanted, ingestedCount.get());
+            int index;
+            while ((index = queryIndex.addAndGet(batchSize)) <= totalWanted) {
+                final int i = index;
+
+                executorService.submit(() -> {
+                    final SourceResponse response = query(queryFacade,
+                            filter,
+                            i,
+                            getQuerySizeFromIndex(totalWanted, i));
+                    ingestMetacards(ingestFacade, getMetacardsFromSourceResponse(response));
+                    printProgressAndFlush(start, totalWanted, ingestedCount.get());
+                });
+            }
+
+            executorService.shutdown();
+
+            while (!executorService.isTerminated()) {
+                try {
+                    TimeUnit.SECONDS.sleep(1);
+                } catch (InterruptedException e) {
+                    // ignore
+                }
+            }
+        } else {
+            while (queryIndex.addAndGet(batchSize) <= totalWanted) {
+                printProgressAndFlush(start, totalWanted, ingestedCount.get());
+
+                final SourceResponse response = query(queryFacade,
+                        filter,
+                        queryIndex.get(),
+                        getQuerySizeFromIndex(totalWanted, queryIndex.get()));
+                ingestMetacards(ingestFacade, getMetacardsFromSourceResponse(response));
+            }
+        }
+
+        printProgressAndFlush(start, totalWanted, ingestedCount.get());
+
+        if (failedCount.get() > 0) {
+            LOGGER.info("Not all records were ingested. [{}] failed", failedCount.get());
+            if (StringUtils.isNotBlank(failedDir)) {
+                try {
+                    writeFailedMetacards(failedMetacards);
+                } catch (IOException e) {
+                    console.println("Error occurred while writing failed metacards to failedDir.");
+                }
+            }
+        }
+    }
+
+    /**
+     * On the final iteration of the loop when the maxMetacards is less than the number of metacards
+     * available in the ingestFacade, the query should only return the remaining wanted metacards,
+     * not the full batch size.
+     *
+     * @param totalPossible - the total hits from the ingestFacade or the maxMetacards
+     * @param index         - the index to be queried next
+     * @return how many metacards should be returned by a query starting at {@param index}
+     */
+    private long getQuerySizeFromIndex(final long totalPossible, final long index) {
+        if (totalPossible - (index - 1) < batchSize) {
```

Comment:
Try https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#min-int-int- for improved readability
`Math.min(totalPossible` - (index - 1), batchSize);`